src/virtual_file.c

```diff
@@ -53,6 +53,14 @@ size_t VFile_GetPos(const VFILE *file)
     return file->cur_ptr - file->content;
 }
 
+void VFile_SetPos(VFILE *file, const size_t pos)
+{
+    const size_t cur_pos = VFile_GetPos(file);
+    const int32_t offset = pos - cur_pos;
+    assert(cur_pos + offset <= file->size);
+    file->cur_ptr += offset;
```

Comment:
Maybe we could just go with `file->cur_ptr = file->content + pos`, and add an assertion that `pos >= 0`?

Reply:
Fixed, thanks. Left the assert doing an upper check as `size_t` is unsigned (produces a warning adding it).